File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.java
class Solution {
     public int maxDistToClosest(int[] seats) {
        if (seats == null || seats.length == 0) {
            return 0;
        }
        int prev = -1;
        int max = 0;
        for (int i = 0; i < seats.length; i++) {
            if (seats[i] == 1) {
                if (prev == -1) {
                    max = i;
                } else {
                    max = Math.max(max, (i - prev) / 2);
                }
                prev = i;
            }
        }
        max = Math.max(max, seats.length - 1 - prev);
        return max;
    }
}

File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int lastind = -1, ans = 0;
        int n = seats.size();
        for(int i = 0; i < n; i++){
            if(seats[i] == 1){
                if(lastind == -1)
                    ans = i;
                else ans = max(ans, (i - lastind)/2);
                lastind = i;
            }
        }
        return max(ans, n-1-lastind);
    }
};

